### PR TITLE
Apply defaults(if any) if column undefined in mongo + Typo Fix

### LINKED ISSUE #79: v3Mongo->v3SQL - Default values not applied to tables when attributes undefined
## Bug report
This issue is only impactful because:
The properties field in 'strapi_permission' is not-required, and has a default value of {}, so realistically everyone should have this in their MongoDB. However, some may have migrated from older versions, and only have the 'fields' property. This could also silently cause other issues with long-time mongo databases being migrated

### Required System information

- Node.js version: v14.21.1
- NPM version: v9.1.2
- Source Strapi version: 3.6.10 
- Target Strapi version: 3.6.10
- Source Database: MongoDB
- Target Database: PostgreSQL
- Operating system: Mac
- Which script are you running: Mongov3->SQLv3 

### Describe the bug

When an attribute that has a default value is undefined, and the migration assistant copies it to the SQL, it will be copied as NULL.
Inside the SQL, there is no 'DEFAULT' specified for columns that have a 'default' value in the core_store entry

### Steps to reproduce the behaviour

1. Go to your MongoDB (Strapi v3) database editor (MongoDB Compass)
2. Go to 'strapi_permission' collection
3. Choose an entry with a properties field, and delete that attribute.
4. Save/Replace/Update your changes on that entry
5. Run the migration assistant
6. Query `SELECT * FROM strapi_permission WHERE properties IS NULL;`
7. You will have at least 1 entry

#### Additional steps for why this impacts Strapi:
1. Run `yarn install && yarn develop` in your app directory
2. The run will fail & exit with code 1 - this is directly related to the 'properties' field being 'null' instead of '{}' (default value)
Note: While this is unlikely, this could cause more hidden issues for other columns elsewhere that expect a default value.

### Expected behaviour

Entries which have default values in core_store, will have their default values inserted into the SQL if it is otherwise left NULL during insertion.

### Code snippets
Strapi_permission core_store structure - Note: relevant fields are 'attributes>properties|conditions', which has 'default: {}'
```{
   "uid":"strapi::permission",
   "collectionName":"strapi_permission",
   "kind":"collectionType",
   "info":{
      "name":"Permission",
      "description":""
   },
   "options":{
      "timestamps":[
         "createdAt",
         "updatedAt"
      ]
   },
   "pluginOptions":{
      "content-manager":{
         "visible":false
      },
      "content-type-builder":{
         "visible":false
      }
   },
   "attributes":{
      "action":{
         "type":"string",
         "minLength":1,
         "configurable":false,
         "required":true
      },
      "subject":{
         "type":"string",
         "minLength":1,
         "configurable":false,
         "required":false
      },
      "properties":{
         "type":"json",
         "configurable":false,
         "required":false,
         "default":{
            
         }
      },
      "conditions":{
         "type":"json",
         "configurable":false,
         "required":false,
         "default":[
            
         ]
      },
      "role":{
         "configurable":false,
         "model":"role",
         "plugin":"admin"
      }
   }
}
```

### Additional context
Note: This error only applies to long-term mongoDB users who's data has persisted long enough to not have a 'properties' field in strapi_permissions/have the old 'fields' column, and are attempting to migrate. This issue is for when an attribute does not have a 'non-required' attribute specified (that has a default value)

For long-time mongoDB users, they might have production databases without properties, or have the old 'fields' attribute. 
When the properties field in mongoDB is undefined, it is migrated as null to SQL. Null values properties cause 'Type Error: Cannot convert undefined or null to object' error in `/Path/To/Strapi/app/node_modules/strapi-admin/services/role.js:51:17`

### PR DESCRIPTION
**Pull request by Skwangles**
fixes #79 
If undefined, checks for default, if found, adds it - i.e. if mongoDB entry doesn't have a column value that it should have, add its default (if it exists). This problem is really strapi v3's problem, where it does not add DEFAULT properties to tables (specifically PostgreSQL)
Also, fixed an obvious typo.

### Why this approach
Since `Object.keys(entry).forEach(` already checks if a property is in the `model.attributes` (then returns if not),
I found it would work just as good the other way, to loop through the model's attributes and check if the entry contains it (trys to apply the default value if doesn't). 

### My Approach
I have only implemented the case where an `attribute.default` exists and the property is UNDEFINED (i.e. does not exist, **not** `= null`), the logic being that if it is '= null' it could have been set null for a reason, thus should only do this for undefined.
*Note: When I say 'undefined', I mean it is physically not defined in the object (i.e. 'b' is "undefined" in {a:1})*

I have limited testing capacity, so would appreciate some further testing on other people's systems for a wider range of applications.

### CODE REVIEW

**Comment by Xander-SquareKicker** [comment]
Applying defaults to undefined properties could be dangerous, as uncleaned/old datasets could just 'work' without being properly cleaned - i.e. uses 'fields' properties instead of the 'properties' attribute, the properties attribute with this will just be applied empty (default), without the 'fields' entries being switched over manually by a developer, causing unseen issues?

tldr; May not be a a good idea to apply defaults, if a developer might need to resolve it manually
